File: src/main/java/com/example/demo/entity/Opinion.java
package com.example.demo.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
public class Opinion {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int idopinion;
	private int iduseropinion;
	private int score;
	private String comment;
}

File: lib/models/opinion.dart
class Opinion {
  final int idOpinion; 
  final int idUserOpinion;
  final int score;
  final String comment;

  Opinion({
    required this.idOpinion,
    required this.idUserOpinion,
    required this.score,
    required this.comment,
  });

  factory Opinion.fromJson(Map<String, dynamic> json) {
    return Opinion(
      idOpinion: json['idopinion'],
      idUserOpinion: json['iduseropinion'],
      score: json['score'],
      comment: json['comment'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idopinion': idOpinion,
      'iduseropinion': idUserOpinion,
      'score': score,
      'comment': comment,
    };
  }
}
